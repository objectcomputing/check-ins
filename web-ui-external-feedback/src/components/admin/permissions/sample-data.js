const allPermissions = [
  'View Action Items',
  'Edit Action Items',
  'Create Action Items',
  'Add Team Members',
  'Delete Team Members'
];

const admin = {
  id: 1,
  name: 'Admin',
  permissions: [
    'View Action Items',
    'Edit Action Items',
    'Create Action Items',
    'Add Team Members',
    'Delete Team Members'
  ]
};

const PDL = {
  id: 2,
  name: 'PDL',
  permissions: [
    'View Action Items',
    'Edit Action Items',
    'Create Action Items',
    'Add Team Members'
  ]
};

const teamLead = {
  id: 3,
  name: 'Team Lead',
  permissions: ['View Action Items', 'Add Team Members']
};

const teamMembers = {
  id: 4,
  name: 'Team Member',
  permissions: ['View Action Items']
};

const roles = [admin, PDL, teamLead, teamMembers];

const handleChange = (role, permission) => {
  alert(`toggle the permission titled "${permission}" for role: ${role.name}`);
  console.log(role);
  console.log(permission);
};

export { allPermissions, roles, handleChange };
