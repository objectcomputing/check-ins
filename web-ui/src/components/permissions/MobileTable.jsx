import {  
  Paper, 
  Table, 
  TableBody, TableCell, 
  TableRow, 
  TableHead, 
  TableContainer,
  Checkbox,
  IconButton,
  TextField,
} from '@material-ui/core'
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import React, { useState } from 'react'
import Autocomplete from '@material-ui/lab/Autocomplete';


export default function PermissionsTableMobile({roles, allPermissions, handleChange}) {
  const [roleIndex, setRoleIndex] = useState(0);
  const [searchPermission, setSearchPermission] = useState("");


  const selectedRole = roles[roleIndex];
  const matchedPermission = searchPermission? 
    allPermissions.find(permission => permission === searchPermission) 
    : 
    null;

  const nextRole = () => {
    if (roleIndex !== roles.length -1){
      setRoleIndex(prevIndex => prevIndex + 1)
    }
  }

  const prevRole = () => {
    if (roleIndex !== 0){
      setRoleIndex(prevIndex => prevIndex -1)
    }
  }


  return (
    <TableContainer component={Paper}>
      <Table aria-label="simple table">   
        <TableHead>
          <TableRow>
            <TableCell>
              <Autocomplete
                size="small"
                style={{minWidth: 200, maxWidth: 350}}
                id="permissions-autocomplete"
                onChange={(event, value) => setSearchPermission(value)}
                options={allPermissions}
                getOptionLabel={(option) => option}
                renderInput={(params) => <TextField {...params} label="Find a permission" variant="outlined" />}
              />
            </TableCell>
            <TableCell align="right" padding="none">
              <IconButton 
                size="small" 
                aria-label="arrow-backward"
                disabled={roleIndex === 0? true: false}
                onClick={prevRole}  
              >
                <ArrowBackIosIcon />
              </IconButton>
              <IconButton 
                size="small" 
                aria-label="arrow-foward"
                disabled={roleIndex === roles.length - 1? true: false}
                onClick={nextRole}
              >
                <ArrowForwardIosIcon />
              </IconButton>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableHead>
          <TableRow>
            <TableCell style={{fontSize: '1.1rem'}}>Permissions</TableCell>
              <TableCell 
                key={selectedRole.id} 
                style={{fontSize: '1.1rem', width: "140px"}} 
                align="right">
                  {selectedRole.name}
              </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {matchedPermission?
            <TableRow key={matchedPermission}>
              <TableCell component="th" scope="row">
                {matchedPermission}
              </TableCell>
              <TableCell key={selectedRole.id} align="right">
                <Checkbox 
                  checked={selectedRole.permissions.indexOf(matchedPermission) !== -1? true : false}
                  onChange={() => handleChange(selectedRole, matchedPermission)}
                  inputProps={{ 'aria-label': 'primary checkbox' }}
                />
              </TableCell>
            </TableRow>
          :
            allPermissions.map(permission => (
              <TableRow key={permission}>
                <TableCell component="th" scope="row">
                  {permission}
                </TableCell>
                <TableCell key={selectedRole.id} align="right">
                    <Checkbox 
                      checked={selectedRole.permissions.indexOf(permission) !== -1? true : false}
                      onChange={() => handleChange(selectedRole, permission)}
                      inputProps={{ 'aria-label': 'primary checkbox' }}
                    />
                  </TableCell>
              </TableRow>
              )
            )
          }
        </TableBody>
      </Table>
    </TableContainer>
  )
}

