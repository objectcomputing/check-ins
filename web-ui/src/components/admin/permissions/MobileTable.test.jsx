import React from 'react';
import MobileTable from './MobileTable';
import { roles, allPermissions, handleChange } from './sample-data';

describe('MobileTable', () => {
  it('renders correctly', () => {
    snapshot(
      <MobileTable
        roles={roles}
        allPermissions={allPermissions}
        handleChange={handleChange}
      />
    );
  });
});
