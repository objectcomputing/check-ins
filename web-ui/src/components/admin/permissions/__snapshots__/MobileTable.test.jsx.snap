// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileTable > renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-ia8vjw-MuiAutocomplete-root"
              style="min-width: 200px; max-width: 350px;"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="permissions-autocomplete"
                  id="permissions-autocomplete-label"
                >
                  Find a permission
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id="permissions-autocomplete"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Find a permission
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-paddingNone MuiTableCell-sizeMedium css-m1xwsv-MuiTableCell-root"
            scope="col"
          >
            <button
              aria-label="arrow-backward"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIosIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
            </button>
            <button
              aria-label="arrow-foward"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowForwardIosIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </th>
        </tr>
      </thead>
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
            style="font-size: 1.1rem;"
          >
            Permissions
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
            scope="col"
            style="font-size: 1.1rem; width: 140px;"
          >
            Admin
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            View Action Items
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Edit Action Items
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Create Action Items
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Add Team Members
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            Delete Team Members
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
